packages/esm-sms-app/src/providers/modals/config-upload-modal.component.tsx

```diff
@@ -0,0 +1,169 @@
+        return uploadConfigTemplate(fileToUpload)
+          .then(() => {
+            mutateTemplates?.();
+            showSnackbar({
+              title: t('uploadSuccess', 'Upload successful'),
+              kind: 'success',
+            }),
+              closeModal();
+          })
+          .catch(() =>
+            showSnackbar({
+              title: t('uploadNotSuccess', 'Upload unsuccessful'),
+              kind: 'error',
+            }),
+          );
+      }
+    },
+    [closeModal, fileToUpload, mutateTemplates, t],
+  );
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <ModalHeader closeModal={onCancel} title={t('uploadConfig', 'Upload configuration template')} />
+      <ModalBody className={styles.modalBody}>
+        {fileToUpload ? (
+          <>
+            <div className={styles.filePlaceholder}>
+              <DocumentUnknown size={16} />
+            </div>
+            <div className={styles.imageDetails}>
+              <Stack gap={5}>
+                <div className="cds--label">{t('fileName', 'File name')}</div>
+                <h6>{fileToUpload.name}</h6>
+                {fileContent && (
+                  <div className={styles.jsonPreviesw}>
+                    <TextArea labelText={t('fileContent', 'File Content')} value={fileContent} readOnly rows={10} />
+                  </div>
+                )}
+              </Stack>
+            </div>
+          </>
+        ) : (
+          <div className="cds--file__container">
+            {errorNotification && (
+              <div className={styles.errorContainer}>
+                <InlineNotification
+                  aria-label="Upload error notification"
+                  kind="error"
+                  onClose={() => setErrorNotification(null)}
+                  subtitle={errorNotification.subtitle}
+                  title={errorNotification.title}
+                />
+              </div>
+            )}
+            <p className="cds--label-description">
+              {t('fileUploadSizeConstraints', 'Size limit is {{fileSize}}MB', {
+                fileSize: maxFileSize,
+              })}
+              .
+            </p>
+            <div className={styles.uploadFile}>
+              <FileUploaderDropContainer
+                accept={['.json']}
+                labelText={t('fileSizeInstructions', 'Drag and drop files here or click to upload')}
+                tabIndex={0}
+                onAddFiles={(evt, { addedFiles }) => {
+                  upload(addedFiles);
+                }}
+              />
+            </div>
+          </div>
+        )}
+      </ModalBody>
+      <ModalFooter>
+        <Button kind="secondary" size="lg" onClick={onCancel}>
+          {t('cancel', 'Cancel')}
+        </Button>
+        <Button type="submit" size="lg" disabled={!fileToUpload}>
+          {t('uploadConfig', 'Upload config')}
+        </Button>
+      </ModalFooter>
+    </Form>
+  );
+};
+
+export default MediaUploaderComponent;
```

Comment:
The component name should be the same as the file name. So here, `ConfigUploadModal`.